src/rust/inetstack/protocols/tcp/peer.rs

```diff
@@ -170,31 +170,38 @@ impl<const N: usize> TcpPeer<N> {
         Ok(new_qd)
     }
 
-    pub fn bind(&self, qd: QDesc, mut addr: SocketAddrV4) -> Result<(), Fail> {
-        let mut inner: RefMut<Inner<N>> = self.inner.borrow_mut();
+    pub fn bind(&self, qd: QDesc, local: SocketAddrV4) -> Result<(), Fail> {
+        // Check if we are binding to the wildcard address.
+        // FIXME: https://github.com/demikernel/demikernel/issues/189
+        if local.ip() == &Ipv4Addr::UNSPECIFIED {
+            let cause: String = format!("cannot bind to wildcard address (qd={:?})", qd);
+            error!("bind(): {}", cause);
+            return Err(Fail::new(libc::ENOTSUP, &cause));
+        }
 
-        // Check if address is already bound.
-        for (socket_id, _) in &inner.addresses {
-            match socket_id {
-                SocketId::Passive(local) | SocketId::Active(local, _) if *local == addr => {
-                    return Err(Fail::new(libc::EADDRINUSE, "address already in use"))
-                },
-                _ => (),
-            }
+        // Check if we are binding to the wildcard port.
+        // FIXME: https://github.com/demikernel/demikernel/issues/582
```

Comment:
What is the reason for not allocating an ephemeral port here?

Reply:
To match same logic of other LibOSes. Eventually it will be unified.